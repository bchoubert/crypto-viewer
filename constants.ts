export const SETTINGS_KEYS = {
  QUOTE_STORAGE_KEY: 'CV_QUOTE_STORAGE_KEY',
  DATE_FORMAT_KEY: 'CV_DATE_FORMAT_KEY',
  WALLET_KEY: 'CV_WALLET_KEY',
  FAVOURITES_KEY: 'CV_FAVOURITES_KEY',
  GRAPH_MODE_KEY: 'CV_GRAPH_MODE_KEY',
  LANGUAGE: 'CV_LANGUAGE',
  DARK_MODE_KEY: 'CV_DARK_MODE_KEY',
};

export type SettingsKeysType = keyof typeof SETTINGS_KEYS;
