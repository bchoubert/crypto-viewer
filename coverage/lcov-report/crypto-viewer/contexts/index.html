
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for crypto-viewer/contexts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> crypto-viewer/contexts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50.63% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>40/79</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.38% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>11/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">49.32% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>36/73</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="NavigationProvider.tsx"><a href="NavigationProvider.tsx.html">NavigationProvider.tsx</a></td>
	<td data-value="65.52" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 65%"></div><div class="cover-empty" style="width: 35%"></div></div>
	</td>
	<td data-value="65.52" class="pct medium">65.52%</td>
	<td data-value="29" class="abs medium">19/29</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="62.96" class="pct medium">62.96%</td>
	<td data-value="27" class="abs medium">17/27</td>
	</tr>

<tr>
	<td class="file low" data-value="SettingsProvider.tsx"><a href="SettingsProvider.tsx.html">SettingsProvider.tsx</a></td>
	<td data-value="10.53" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 10%"></div><div class="cover-empty" style="width: 90%"></div></div>
	</td>
	<td data-value="10.53" class="pct low">10.53%</td>
	<td data-value="19" class="abs low">2/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	</tr>

<tr>
	<td class="file low" data-value="ThemeProvider.tsx"><a href="ThemeProvider.tsx.html">ThemeProvider.tsx</a></td>
	<td data-value="47.83" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 47%"></div><div class="cover-empty" style="width: 53%"></div></div>
	</td>
	<td data-value="47.83" class="pct low">47.83%</td>
	<td data-value="23" class="abs low">11/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="22" class="abs low">10/22</td>
	</tr>

<tr>
	<td class="file high" data-value="TranslationProvider.tsx"><a href="TranslationProvider.tsx.html">TranslationProvider.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Sep 06 2021 22:37:01 GMT+0200 (Central European Summer Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    