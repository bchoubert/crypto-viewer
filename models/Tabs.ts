
export type tabType = 'list' | 'wallet' | 'settings' | 'details';

const Tabs = {
  list: 'list' as tabType,
  details: 'details' as tabType,
  settings: 'settings' as tabType,
  wallet: 'wallet' as tabType,
}

export default Tabs;
