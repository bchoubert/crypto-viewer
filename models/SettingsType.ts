import { AvailableTranslations } from '../assets/translations/TranslationUtils';
import { SettingsKeysType } from '../constants';
import dateFormats, { DateFormatType } from './DateFormat';
import { GraphModeType } from './GraphMode';
import { DarkModeType } from './DarkMode';
import QuoteType from './QuoteType';
import WalletItem from './WalletItem';

export type SettingsValue = AvailableTranslations |
QuoteType | DateFormatType | WalletItem[] | GraphModeType | DarkModeType | string[];
type SettingsType = Record<SettingsKeysType, SettingsValue>;

export const settingDetails: Record<SettingsKeysType, {
  needsParsing: boolean,
  defaultValue: SettingsValue,
}> = {
  QUOTE_STORAGE_KEY: { needsParsing: true, defaultValue: { code: 'USD', symbol: '$' } as QuoteType },
  DATE_FORMAT_KEY: { needsParsing: false, defaultValue: dateFormats.american as DateFormatType },
  FAVOURITES_KEY: { needsParsing: true, defaultValue: [] as string[] },
  GRAPH_MODE_KEY: { needsParsing: false, defaultValue: 'Simple' as GraphModeType },
  WALLET_KEY: { needsParsing: true, defaultValue: [] as WalletItem[] },
  LANGUAGE: { needsParsing: false, defaultValue: AvailableTranslations.en },
  DARK_MODE_KEY: { needsParsing: false, defaultValue: 'light' as DarkModeType },
};

export const defaultSettings: SettingsType = Object.keys(settingDetails).reduce((result, key) => ({
  ...result,
  [key]: settingDetails[key].defaultValue,
}), {} as SettingsType);

export default SettingsType;
