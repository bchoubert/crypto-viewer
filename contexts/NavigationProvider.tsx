import React, {
  createContext, FC, ReactNode, useCallback, useState, useEffect, useMemo, memo, useContext,
} from 'react';
import { StatusBarStyle, BackHandler } from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';

import Tabs, { TabType } from '../models/Tabs';
import Crypto from '../models/Crypto';
import ColorService from '../services/Color.service';
import { ThemeContext } from './ThemeProvider';

export const NavigationContext = createContext({
  activeTab: Tabs.list as TabType,
  details: null as Crypto | null,
  statusBarColor: Colors.white as string,
  statusBarMode: 'dark-content' as StatusBarStyle,
  changeTab: null as (tabName: TabType, newDetails?: Object) => void | null,
});

interface NavigationProviderProps {
  children: ReactNode;
}

const NavigationProvider: FC<NavigationProviderProps> = ({ children }) => {
  const theme = useContext(ThemeContext);

  const [activeTab, setActiveTab] = useState<TabType>(Tabs.list);

  // Store details about current active tab
  const [details, setDetails] = useState<Crypto | null>(null);

  // StatusBar mode
  const [statusBarColor, setStatusBarColor] = useState<string>(theme.backgroundColor);
  const [statusBarMode, setStatusBarMode] = useState<StatusBarStyle>('dark-content');

  const changeTab = useCallback((tabName: TabType, newDetails: Crypto = null) => {
    setStatusBarColor(tabName === Tabs.details
      ? ColorService.getColorFromCrypto(newDetails.id)
      : theme.backgroundColor);
    setStatusBarMode(tabName === Tabs.details ? 'light-content' as StatusBarStyle : 'dark-content' as StatusBarStyle);
    setDetails(newDetails);
    setActiveTab(tabName);
  }, []);

  const handleBackButton = useCallback(() => {
    if ([Tabs.settings, Tabs.details].includes(activeTab)) {
      changeTab(Tabs.list);
      return true;
    }
    return false;
  }, [activeTab, changeTab]);

  useEffect(() => {
    const backHandler = BackHandler.addEventListener('hardwareBackPress', handleBackButton);
    return () => backHandler.remove();
  }, [activeTab]);

  const contextValues = useMemo(() => ({
    activeTab,
    details,
    statusBarColor,
    statusBarMode,
    changeTab,
  }), [activeTab, details, statusBarColor, statusBarMode, changeTab]);

  useEffect(() => {
    setStatusBarColor(theme.backgroundColor);
  }, [theme]);

  return (
    <NavigationContext.Provider value={contextValues}>
      {children}
    </NavigationContext.Provider>
  );
};

export default memo(NavigationProvider);
