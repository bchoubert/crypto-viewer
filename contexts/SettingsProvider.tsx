import React, {
  createContext, FC, memo, ReactNode, useState, useEffect, useMemo, useCallback,
} from 'react';
import { Text } from 'react-native';

import { SettingsKeysType } from '../constants';
import SettingsType, { defaultSettings, SettingsValue } from '../models/SettingsType';
import SettingsService from '../services/Settings.service';

export const SettingsContext = createContext({
  settings: defaultSettings as SettingsType,
  changeSettings: null as (key: SettingsKeysType, newValue: SettingsValue) => void | null,
});

interface SettingsProviderProps {
  children: ReactNode;
}

const SettingsProvider: FC<SettingsProviderProps> = ({
  children,
}) => {
  const [isLoading, setLoading] = useState<boolean>(true);
  const [values, setValues] = useState<SettingsType>(defaultSettings);

  useEffect(
    () => {
      const loadSettings = async () => {
        const settings = await SettingsService.loadAll();
        setValues(settings);
        setLoading(false);
      };
      loadSettings();
    },
    [],
  );

  const handleChangeSettings = useCallback(
    async (key: SettingsKeysType, newValue: SettingsValue) => {
      await SettingsService.changeSetting(key, newValue);
      setValues((v) => ({
        ...v,
        [key]: newValue,
      }));
    },
    [],
  );

  const contextValues = useMemo(() => ({
    settings: values,
    changeSettings: handleChangeSettings,
  }), [values, handleChangeSettings]);

  if (isLoading) {
    return <Text>Loading...</Text>;
  }

  return (
    <SettingsContext.Provider value={contextValues}>
      {children}
    </SettingsContext.Provider>
  );
};

export default memo(SettingsProvider);
