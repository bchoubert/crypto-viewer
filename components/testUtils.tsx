/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderAPI } from '@testing-library/react-native';
import '@testing-library/jest-dom/extend-expect';
import React, { ReactElement } from 'react';

import NavigationProvider from '../contexts/NavigationProvider';
import { SettingsContext } from '../contexts/SettingsProvider';
import ThemeProvider from '../contexts/ThemeProvider';
import TranslationProvider from '../contexts/TranslationProvider';

import { defaultSettings } from '../models/SettingsType';

jest.useFakeTimers();

export const enrichProviders = (node: ReactElement): ReactElement => (
  <SettingsContext.Provider
    value={{
      settings: defaultSettings,
      changeSettings: jest.fn(),
    }}
  >
    <TranslationProvider>
      <ThemeProvider>
        <NavigationProvider>
          {node}
        </NavigationProvider>
      </ThemeProvider>
    </TranslationProvider>
  </SettingsContext.Provider>
);

// eslint-disable-next-line max-len
export const renderNode = (node: ReactElement): RenderAPI => render(enrichProviders(node));
