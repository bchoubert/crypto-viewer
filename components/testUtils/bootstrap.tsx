/* eslint-disable no-console */
/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderAPI } from '@testing-library/react-native';
import '@testing-library/jest-dom/extend-expect';
import React, { ReactElement } from 'react';
import * as PassThoughComponents from '../PassThroughComponents';

import NavigationProvider from '../../contexts/NavigationProvider';
import { SettingsContext } from '../../contexts/SettingsProvider';
import ThemeProvider from '../../contexts/ThemeProvider';
import TranslationProvider from '../../contexts/TranslationProvider';

import { defaultSettings } from '../../models/SettingsType';
import consoleMocker from './consoleMocker';

jest.useFakeTimers();

export const enrichProviders = (node: ReactElement): ReactElement => (
  <SettingsContext.Provider
    value={{
      settings: defaultSettings,
      changeSettings: jest.fn(),
    }}
  >
    <TranslationProvider>
      <ThemeProvider>
        <NavigationProvider>
          {node}
        </NavigationProvider>
      </ThemeProvider>
    </TranslationProvider>
  </SettingsContext.Provider>
);

// eslint-disable-next-line max-len
export const renderNode = (node: ReactElement): RenderAPI => render(enrichProviders(node));

beforeAll(() => {
  consoleMocker.mockConsole();
  // eslint-disable-next-line no-extend-native
  Date.prototype.getTimezoneOffset = jest.fn(() => 0);
});

afterAll(() => {
  consoleMocker.restoreConsole();
  jest.resetAllMocks();
});

jest.spyOn(PassThoughComponents, 'Image');
(PassThoughComponents.Image as unknown as jest.Mock).mockImplementation(() => (<div>IMAGE</div>));
