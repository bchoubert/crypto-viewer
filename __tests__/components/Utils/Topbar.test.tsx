import { fireEvent } from '@testing-library/react-native';
import React from 'react';
import { renderNode } from '../../../components/testUtils/bootstrap';
import Topbar from '../../../components/Utils/Topbar';
import { SettingsKeysType } from '../../../constants';
import { NavigationContext, NavigationContextInterface } from '../../../contexts/NavigationProvider';
import { SettingsContext } from '../../../contexts/SettingsProvider';
import { defaultSettings, SettingsValue } from '../../../models/SettingsType';
import Tabs from '../../../models/Tabs';

describe('<TopBar />', () => {
  // Remove errors from toast android
  // eslint-disable-next-line no-console
  console.error = () => null;

  const changeTabFn = jest.fn();
  const changeSettingsFn = jest.fn();

  const commonNavigationContextValue: NavigationContextInterface = {
    activeTab: Tabs.list,
    changeTab: changeTabFn,
    handleBackAction: () => null,
    statusBarColor: '#334455',
    statusBarMode: 'default',
    details: null,
  };

  const navigationDetailsValue = {
    id: 'btc',
    name: 'Bitcoin',
    price: 2,
    status: 'online' as 'online',
    details: {
      type: '',
      symbol: 'B',
    },
  };

  const commonSettingsContextValue = {
    changeSettings: changeSettingsFn,
    settings: defaultSettings,
  };

  it('Prints correctly', () => {
    const { toJSON } = renderNode(
      <NavigationContext.Provider value={commonNavigationContextValue}>
        <Topbar />
      </NavigationContext.Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('List behaviour', () => {
    const node = renderNode(
      <NavigationContext.Provider value={commonNavigationContextValue}>
        <Topbar />
      </NavigationContext.Provider>,
    );

    node.getByTestId('logo');
    node.getByTestId('settings');
  });

  it('Settings behaviour', () => {
    const node = renderNode(
      <NavigationContext.Provider
        value={{
          ...commonNavigationContextValue,
          activeTab: Tabs.settings,
        }}
      >
        <Topbar />
      </NavigationContext.Provider>,
    );

    node.getByTestId('backFromSettings');
  });

  it('Wallet behaviour', () => {
    const node = renderNode(
      <NavigationContext.Provider
        value={{
          ...commonNavigationContextValue,
          activeTab: Tabs.wallet,
        }}
      >
        <Topbar />
      </NavigationContext.Provider>,
    );

    node.getByTestId('logo');
    node.getByTestId('settings');
  });

  it('Details behaviour', () => {
    const node = renderNode(
      <NavigationContext.Provider
        value={{
          ...commonNavigationContextValue,
          activeTab: Tabs.details,
          details: navigationDetailsValue,
        }}
      >
        <Topbar />
      </NavigationContext.Provider>,
    );

    node.getByTestId('backFromCrypto');
    node.getByTestId('favourites');
    node.getByTestId('settings');
  });

  it('Add to favourites', () => {
    const node = renderNode(
      <SettingsContext.Provider value={commonSettingsContextValue}>
        <NavigationContext.Provider
          value={{
            ...commonNavigationContextValue,
            activeTab: Tabs.details,
            details: navigationDetailsValue,
          }}
        >
          <Topbar />
        </NavigationContext.Provider>
      </SettingsContext.Provider>,
    );

    node.getByTestId('favourites');
    node.getByTestId('favourites-text');

    expect(node.getByTestId('favourites-text').props.accessibilityLabel).toEqual('Add to favourites');

    expect(changeSettingsFn).toHaveBeenCalledTimes(0);

    fireEvent.press(node.getByTestId('favourites'));

    expect(changeSettingsFn).toHaveBeenCalledTimes(1);
    expect(changeSettingsFn).toHaveBeenCalledWith('FAVOURITES_KEY' as SettingsKeysType, ['btc']);
  });

  it('Remove from favourites', () => {
    const node = renderNode(
      <SettingsContext.Provider
        value={{
          ...commonSettingsContextValue,
          settings: {
            ...commonSettingsContextValue.settings,
            FAVOURITES_KEY: ['btc'] as SettingsValue,
          },
        }}
      >
        <NavigationContext.Provider
          value={{
            ...commonNavigationContextValue,
            activeTab: Tabs.details,
            details: navigationDetailsValue,
          }}
        >
          <Topbar />
        </NavigationContext.Provider>
      </SettingsContext.Provider>,
    );

    node.getByTestId('favourites');
    node.getByTestId('favourites-text');

    expect(node.getByTestId('favourites-text').props.accessibilityLabel).toEqual('Remove from favourites');

    expect(changeSettingsFn).toHaveBeenCalledTimes(1);

    fireEvent.press(node.getByTestId('favourites'));

    expect(changeSettingsFn).toHaveBeenCalledTimes(2);
    expect(changeSettingsFn).toHaveBeenCalledWith('FAVOURITES_KEY' as SettingsKeysType, []);
  });
});
