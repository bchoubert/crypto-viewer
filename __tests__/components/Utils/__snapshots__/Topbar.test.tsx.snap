// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> Prints correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexBasis": 60,
      "flexDirection": "row",
      "flexGrow": 0,
      "justifyContent": "space-between",
      "paddingLeft": 8,
      "paddingRight": 8,
      "position": "relative",
      "zIndex": 3,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
    testID="logo"
  >
    <div>
      IMAGE
    </div>
    <Text
      style={
        Object {
          "color": "#005BE3",
          "fontSize": 20,
          "marginLeft": 5,
        }
      }
    >
      Crypto Viewer
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="settings"
  >
    <Text
      style={
        Object {
          "color": "#222222",
          "fontFamily": "crypto-viewer",
          "fontSize": 20,
          "padding": 10,
        }
      }
    >
      Ó†ç
    </Text>
  </View>
</View>
`;
